source/gui/addonGui.py

```diff
@@ -303,16 +309,19 @@ def getAddonStatus(self, addon):
 			# Translators: The status shown for a newly installed addon before NVDA is restarted.
 			statusList.append(_("Install"))
 		# in some cases an addon can be expected to be disabled after install, so we want "install" to take precedence here
-		elif globalVars.appArgs.disableAddons or addon.isDisabled:
+		# If add-ons are globally disabled, don't show this status.
+		elif addon.isDisabled and not globalVars.appArgs.disableAddons:
 			# Translators: The status shown for an addon when its currently suspended do to addons being disabled.
 			statusList.append(_("Disabled"))
 		if addon.isPendingRemove:
 			# Translators: The status shown for an addon that has been marked as removed, before NVDA has been restarted.
 			statusList.append(_("Removed after restart"))
-		elif addon.isPendingDisable or (not addon.isPendingEnable and addon.isPendingInstall and addon.isDisabled):
+		elif addon.isPendingDisable or (addon.isDisabled and not addon.isPendingEnable
+			and (addon.isPendingInstall or globalVars.appArgs.disableAddons)):
 			# Translators: The status shown for an addon when it requires a restart to become disabled
 			statusList.append(_("Disabled after restart"))
-		elif addon.isPendingEnable or (addon.isPendingInstall and not addon.isDisabled):
+		elif addon.isPendingEnable or (not addon.isDisabled

```

Comment:
Also here, slightly longer for clarity, with comments:
```suggestion
		elif addon.isPendingEnable or (
				# yet to be installed, enabled after install
				addon.isPendingInstall and not addon.isDisabled
			) or (
				# addons globally disabled, enabled after restart
				globalVars.appArgs.disableAddons and not addon.isDisabled
			):
```